src/db.rs

```diff
@@ -38,315 +39,338 @@ impl Default for Connection {
     }
 }
 
-pub fn get_dbconn() -> Result<PooledConn> {
+pub async fn get_dbpool() -> Result<AnyPool> {
+    let db_type = env::var("KBS_DB_TYPE").expect("KBS_DB_TYPE not set");
     let host_name = env::var("KBS_DB_HOST").expect("KBS_DB_HOST not set");
     let user_name = env::var("KBS_DB_USER").expect("KBS_DB_USER not set.");
     let db_pw = env::var("KBS_DB_PW").expect("KBS_DB_PW not set.");
     let data_base = env::var("KBS_DB").expect("KBS_DB not set");
 
-    let opts = OptsBuilder::new()
-        .ip_or_hostname(Some(host_name))
-        .user(Some(user_name))
-        .pass(Some(db_pw))
-        .db_name(Some(data_base));
+    let db_url = format!(
+        "{}://{}:{}@{}/{}",
+        db_type, user_name, db_pw, host_name, data_base
+    );
 
-    let dbpool = Pool::new(opts)?;
-    let dbconn = dbpool.get_conn()?;
-    Ok(dbconn)
+    let db_pool = AnyPoolOptions::new()
+        .max_connections(1000)
+        .connect(&db_url)
+        .await
+        .map_err(|e| anyhow!("Encountered error trying to create a mysql pool: {}", e))?;
+    Ok(db_pool)
 }
 
-pub fn insert_connection(connection: Connection) -> Result<Uuid> {
-    let mut dbconn = get_dbconn()?;
+pub async fn insert_connection(connection: Connection) -> Result<Uuid> {
+    let dbpool = get_dbpool().await?;
 
     let nwuuid = Uuid::new_v4();
     let uuidstr = nwuuid.as_hyphenated().to_string();
 
-    let mqstr = "INSERT INTO conn_bundle (id, policy, fw_api_major, fw_api_minor,
-                 fw_build_id, launch_description, fw_digest,create_date)
-                 VALUES(?, ?, ?, ?, ?, ?, ?,NOW())"
-        .to_string();
-
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-
-    trnsx.exec_drop(
-        mqstr,
-        (
-            uuidstr,
-            connection.policy,
-            connection.fw_api_major,
-            connection.fw_api_minor,
-            connection.fw_build_id,
-            connection.launch_description,
-            connection.fw_digest,
-        ),
-    )?;
-    trnsx.commit()?;
+    let query = format!("insert into conn_bundle (id, policy, fw_api_major, fw_api_minor, fw_build_id, launch_description, fw_digest, create_date) VALUES(\"{}\", {}, {}, {}, {}, \"{}\", \"{}\", NOW())", 
+     uuidstr,
+     connection.policy,
+     connection.fw_api_major,
+     connection.fw_api_minor,
+     connection.fw_build_id,
+     connection.launch_description,
+     connection.fw_digest
+     );
```

Comment:
Overall, I agree, but any is written to accept strings.  Mysql uses ?, postgres uses $, and others use a different mechanism for substitutions.  I wasn't able to find a query check for any, but if one does not exist, I can write something to make sure that we are not susceptible to sql injection.

Reply:
It seems sqlx has QueryBuilder, see the [example](https://docs.rs/sqlx/latest/sqlx/struct.QueryBuilder.html#method.push_values). `push_bind()` can insert a single `?` (or `$1` in postgres).

Hopefully it can help (it also explains about SQL injection...)

Reply:
I looked at this early in my development, but it presents challenges for writing a lot of if or match statements based upon the database back-end.  If i use mysql, I would use ?, but if I use postgres, I would need to use $, but this defeats the purpose of using any.  How do we avoid having the equivalent of #ifdef for use statements and for if/match based upon which back-end we are using?  We can write a decent sql injection checker while maintaining general code. I am willing to go either way, I just don't want to put us into a a bad spot of having to maintain a lot of branched code based upon which back end we are using.

Reply:
If I read the docs correctly, calling `.push_bind()` will insert the correct "bind-symbol" for the backend (so on mysql it'll insert `?` and on postgres it'll insert '$1'). So supposedly our code won't have conditions (if/match/#ifdef) on the DB-type.

But, we'll need to build the queries using sqlx's QueryBuilder.

Your approach ("write a decent sql injection checker") means, if I understand correctly, escaping each value we insert to the DB. So that means -- writing the correct escaping function (isn't this different between DB types?) and remembering to call it on each value when added to the statements. If you feel safer with this, we can try. I thought that would be the job of a db-library (maybe sqlx's ergonomics don't fit us).

Reply:
Maybe we are thinking of something different.  Escaping passwords or user names is one level, but I thought that the real importance of checking for sql injection is to prevent people from inserting a semi-colon followed by a DROP database, or 1=1; followed by alter passwords, etc.    

   I will take another look at the QueryBuilder. It is entirely possible that I missed something.  When I first looked at this, the thing that gave me pause was the implementation definition for query builder:
 pub fn push_bind<T>(&mut self, value: T) -> &mut Self
    where
        T: 'args + Encode<'args, DB> + Send + Type<DB>,
    For sqlx::any, some of these traits are not implemented, so I punted and went what seemed like a simpler way by using format! on strings to build up a query. 

    Before we give up, I will write some simple functions using the existing database and QueryBuilder to see if I can easily use the env var KBS_DB_TYPE to build queries.  This might let us realize the benefits of QueryBuilder while still maintaining the generic approach that we wanted from sqlx.  I am not sure that QueryBuilder will prevent malicious strings from getting into the query like ";DROP DATABASE sev_attest", but we can search query args prior to binding and log failures.